lib/phoenix/endpoint.ex

```diff
@@ -388,11 +389,25 @@ defmodule Phoenix.Endpoint do
 
   @doc false
   defmacro __before_compile__(env) do
+    sockets = Module.get_attribute(env.module, :phoenix_sockets)
     plugs = Module.get_attribute(env.module, :plugs)
     {conn, body} = Plug.Builder.compile(env, plugs, [])
 
+    socket_intercepts = for {path, module} <- sockets do
```

Comment:
The paths matching needs to be compiled as a cowboy route or something similar if we want to avoid extra processes in http2. but we will cross that bridge when it comes. Just leave a todo. :)